test/libyul/yulOptimizerTests/fullSuite/no_move_loop_orig.yul

```diff
@@ -12,10 +12,10 @@
 //
 // {
 //     {
-//         let _1 := iszero(caller())
 //         for { }
 //         true
 //         {
+//             let _1 := iszero(caller())
 //             for { } iszero(_1) { }
 //             { }
 //             mstore(192, 0)
```

Comment:
I'm not sure if this tests still tests what it's supposed to test - the comment says that the expression containing `caller()` is not supposed to be movable but the expectations clearly expect it to be moved.

Still, regardless of that the version with `caller()` outside of both loops seems both more optimal and correct so this is a regression. It happens because the new sequence runs `LoopInvariantCodeMotion` (`M`) only once and it happens before inlining. In the default sequence this is not a problem because it has the big loop.

Appending `M` after the inlining segment (`gvif`) fixes the problem. Perhaps this step should always follow inlining.

Reply:
### Comparison with unmodified `the-good-parts-mk2` on `cancun`
#### Tweak
```diff
-       "gvif"                         // Run full inliner
+       "gvifM"                        // Run full inliner
        "CTUca[r]LSsTFOtfDnca[r]Iulc"  // SSA plus simplify

        "scCTUt"
-       "gvif"                         // Run full inliner
+       "gvifM"                        // Run full inliner
        "[scCTUt] TOntnfDIul"          // Perform structural simplification
-       "gvif"                         // Run full inliner
+       "gvifM"                        // Run full inliner
```

#### External tests (`ir-optimize-evm+yul`)
[`seqbench-sequence-the-good-parts-mk2-tweaks-M-after-inlining`](https://app.circleci.com/pipelines/github/ethereum/solidity/33658/workflows/11189344-e383-426e-a2c5-0631aeb07aa5) vs [`seqbench-sequence-the-good-parts-mk2-cancun`](https://app.circleci.com/pipelines/github/ethereum/solidity/33655/workflows/f901ee55-094e-44b1-860c-44cab6b675bb)

|          project |  bytecode_size | deployment_gas |     method_gas |
|:----------------:|---------------:|---------------:|---------------:|
|              ens | **`+0.01% ❌`** |          `-0%` |           `0%` |
|              gp2 | **`+0.03% ❌`** |                |                |
|  perpetual-pools | **`+0.04% ❌`** | **`+0.04% ❌`** |          `-0%` |
|    pool-together | **`-0.04% ✅`** |                |                |
|          uniswap | **`+0.01% ❌`** |                |                |
|         zeppelin |          `-0%` | **`-0.01% ✅`** | **`+0.02% ❌`** |

#### Gas diff stats
| File name                                                       | IR optimized   | Legacy optimized   | Legacy   |
|-----------------------------------------------------------------|----------------|--------------------|----------|
| `array/byte_array_transitional_2.sol`                           | -4%            |                    |          |
| `array/copying/bytes_storage_to_storage.sol`                    | -6%            |                    |          |